StRoot/StDbUtilities/StMagUtilities.cxx

```diff
@@ -2664,8 +2679,10 @@ void StMagUtilities::UndoIFCShiftDistortion( const Float_t x[], Float_t Xprime[]
 void StMagUtilities::UndoSpaceChargeDistortion( const Float_t x[], Float_t Xprime[] , Int_t Sector )
 {
 
-  if ((mDistortionMode & kSpaceCharge) && (mDistortionMode & kSpaceChargeR2)) {
-      cout << "StMagUtilities ERROR **** Do not use kSpaceCharge and kSpaceChargeR2 at the same time" << endl ;
+  if (((mDistortionMode/kSpaceCharge)    & 1) +
+      ((mDistortionMode/kSpaceChargeR2)  & 1) +
+      ((mDistortionMode/kSpaceChargeFXT) & 1) > 1) {
```

Comment:
I would suggest to keep the bitwise & operator as before in order to check the bits against the bit masks. Instead of the logical && you can use the addition + operator to test the mutually exclusive bits

```suggestion
  if ((mDistortionMode & kSpaceCharge) +
      (mDistortionMode & kSpaceChargeR2) +
      (mDistortionMode & kSpaceChargeFXT) > 1) {
```

Reply:
I don't think this works. Each of the kSpaceChargeXXX constants have a value greater than 1, so any of the three being active would result in a sum greater than 1. I'm testing for whether more than one of the three are active.

Reply:
That's right. The bitwise & results in an integer and the comparison of the sum with 1 does not make sense.

Since you expect only one of kSpaceChargeXXX flags to be set then their "union" should give only one bit set to 1, i.e. the number should be a power of 2. So, perhaps something like this can work

```suggestion
  int scopt = (mDistortionMode & kSpaceCharge) |
      (mDistortionMode & kSpaceChargeR2) |
      (mDistortionMode & kSpaceChargeFXT);
  if (scopt & (scopt - 1)) { // more than one specified
```

Just a suggestion.

Reply:
Interesting suggestion, and I believe that works, though one can shorten it to use:

    int scopt = (mDistortionMode & (kSpaceCharge | kSpaceChargeR2 | kSpaceChargeFXT));

It's a subjective call whether that's prettier/clearer....unless....you don't trust `/` as a division operator to return a quotient that is guaranteed not to round up if the remainder is non-zero? As long as it does not round up, it certainly works for cases where the divisor has only a single non-zero bit.  So if you are worried about rounding up, here is an [alternative](https://en.cppreference.com/w/cpp/numeric/math/div) to `/` that must work because it is defined as the quotient (though it is moving towards a less pretty/clear solution):

  if (((std::div(mDistortionMode,kSpaceCharge   )).quot & 1) +
      ((std::div(mDistortionMode,kSpaceChargeR2) ).quot & 1) +
      ((std::div(mDistortionMode,kSpaceChargeFXT)).quot & 1) > 1) {
      ... }

Again, the only reason this could fail is if any of the kSpaceChargeXXX variables has more than one non-zero bit. But there's no reason to explore such cases for this situation.

Perhaps the best reason I can think of to use your solution is that it probably takes fewer CPU cycles, even though this is negligible here. But, I'll go ahead with your solution and be done with this.

-Gene

Reply:
> It's a subjective call whether that's prettier/clearer....unless....you don't trust `/` as a division operator to return a quotient that is guaranteed not to round up if the remainder is non-zero?

Using the division must be fine in this case too since we divide by a power of 2. This is simply equivalent to a certain number of binary shifts to the right. I guess if I saw the bitwise shift operator this piece of code wouldn't even trigger me :)